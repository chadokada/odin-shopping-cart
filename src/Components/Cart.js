import React from 'react';
import '../Styles/App.css';

const Cart = ({showCart, toggleCart}) => {

  let cartOpen = '';
  if (showCart === true){
    cartOpen = ' open';
  }



  return (
    <div className={`cart-drawer${cartOpen}`}>
      THIS IS CARTA
      <button onClick={toggleCart}>Close Cart</button>
    </div>
  )
}

export default Cart;