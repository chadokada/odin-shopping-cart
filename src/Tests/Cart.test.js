import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Cart from '../Components/Cart';

it('should render nothing if cart is empty', () => {
  const cart = {}
  const { renderedCart } = render(<Cart cart={cart}/>)
  expect(renderedCart).toMatchSnapshot()
});

it('should render one item in cart', () => {
  const onInputMock = jest.fn();
  const cart = {
    'Citrus Bag' : {
      name: 'Citrus Bag', 
      price: 45.00,
      source: 'citrus_bag.png', 
      type:'bags',
      quantity: 1
    }
  }
  const { renderedCart } = render(
    <Cart cart={cart} handleQuantityInput={onInputMock}/>
    )

  expect(renderedCart).toMatchSnapshot()
});

it('should render two items in cart', () => {
  const onInputMock = jest.fn();
  const cart = {
    'Citrus Bag' : {
      name: 'Citrus Bag', 
      price: 45.00,
      source: 'citrus_bag.png', 
      type:'bags',
      quantity: 1
    },
    'Bowler' : {
      name: 'Bowler', 
      price: 25.00, 
      source: 'bowler.jpg', 
      type:'hats',
      quantity: 3
    }
  }
  const { renderedCart } = render(
    <Cart cart={cart} handleQuantityInput={onInputMock}/>
    )
    
  expect(renderedCart).toMatchSnapshot()
});

it('should show correct cart total if one item with quantity of 1 in cart', () => {
  const onInputMock = jest.fn();
  const cart = {
    'Citrus Bag' : {
      name: 'Citrus Bag', 
      price: 45.00,
      source: 'citrus_bag.png', 
      type:'bags',
      quantity: 1
    }
  }
  render(<Cart cart={cart} handleQuantityInput={onInputMock}/>)
    
  expect(screen.getByText(
      (content, element) => content.startsWith('Total')).textContent
    ).toMatch('Total: $45.00');
});

it('should show correct cart total if one item with quantity of 2 in cart', () => {
  const onInputMock = jest.fn();
  const cart = {
    'Citrus Bag' : {
      name: 'Citrus Bag', 
      price: 45.00,
      source: 'citrus_bag.png', 
      type:'bags',
      quantity: 2
    }
  }
  render(<Cart cart={cart} handleQuantityInput={onInputMock}/>)
    
  expect(screen.getByText(
      (content, element) => content.startsWith('Total')).textContent
    ).toMatch('Total: $90.00');
});

it('should show correct cart total if multiple items with various quantities in cart', () => {
  const onInputMock = jest.fn();
  const cart = {
    'Citrus Bag' : {
      name: 'Citrus Bag', 
      price: 45.00,
      source: 'citrus_bag.png', 
      type:'bags',
      quantity: 1
    },
    'Bowler' : {
      name: 'Bowler', 
      price: 25.00, 
      source: 'bowler.jpg', 
      type:'hats',
      quantity: 3
    }
  }
  
  render(<Cart cart={cart} handleQuantityInput={onInputMock}/>)
    
  expect(screen.getByText(
      (content, element) => content.startsWith('Total')).textContent
    ).toMatch('Total: $120.00');
});

it('should fire handleCheckOut the correct number of times', () => {
  const onInputMock = jest.fn();
  const onClickMock = jest.fn();
  const cart = {}
  render(
    <Cart 
      cart={cart} 
      handleCheckOut={onClickMock} 
      handleQuantityInput={onInputMock}
    />
  )

  const checkout = screen.getByRole('button', {name: 'Checkout'});
  userEvent.click(checkout);
  
  expect(onClickMock).toHaveBeenCalledTimes(1);
})