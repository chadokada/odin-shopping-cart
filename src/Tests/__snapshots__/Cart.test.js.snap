// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render nothing if cart is empty 1`] = `
<div
  className="cart-drawer"
>
  <div
    className="cart-overlay"
  />
  <div
    className="cart-container"
  >
    <div
      className="cart-header"
    >
      <h2>
        Your Shopping Cart
      </h2>
      <i
        className="fa-solid fa-x"
      />
    </div>
    <div
      className="cart-items-container"
    >
      <div>
        No items yet...
      </div>
    </div>
    <div
      className="cart-total"
    >
      Total: $
      0.00
    </div>
    <button
      className="checkout"
    >
      Checkout
    </button>
  </div>
</div>
`;

exports[`should render one item in cart 1`] = `
<div
  className="cart-drawer"
>
  <div
    className="cart-overlay"
    onClick={[MockFunction]}
  />
  <div
    className="cart-container"
  >
    <div
      className="cart-header"
    >
      <h2>
        Your Shopping Cart
      </h2>
      <i
        className="fa-solid fa-x"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="cart-items-container"
    >
      <div
        className="cart-item"
        name="Citrus Bag"
        type="bags"
      >
        <div
          className="cart-item-img-container"
        >
          <img
            alt={
              Object {
                "name": "Citrus Bag",
                "price": 45,
                "quantity": 1,
                "source": "citrus_bag.png",
                "type": "bags",
              }
            }
            className="cart-item-img"
            src="citrus_bag.png"
          />
        </div>
        <div
          className="cart-item-side"
        >
          <div
            className="cart-item-info"
          >
            <div
              className="cart-item-name"
            >
              Citrus Bag
            </div>
            <div
              className="cart-item-price"
            >
              $45.00
            </div>
          </div>
          <div
            className="cart-item-quantity-container"
          >
            <button
              onClick={[MockFunction]}
            >
              -
            </button>
            <input
              className="cart-item-quantity-input"
              onInput={[MockFunction]}
              type="number"
              value={1}
            />
            <button
              onClick={[MockFunction]}
            >
              +
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="cart-total"
    >
      Total: $
      45.00
    </div>
    <button
      className="checkout"
      onClick={[MockFunction]}
    >
      Checkout
    </button>
  </div>
</div>
`;

exports[`should render two items in cart 1`] = `
<div
  className="cart-drawer"
>
  <div
    className="cart-overlay"
  />
  <div
    className="cart-container"
  >
    <div
      className="cart-header"
    >
      <h2>
        Your Shopping Cart
      </h2>
      <i
        className="fa-solid fa-x"
      />
    </div>
    <div
      className="cart-items-container"
    >
      <div
        className="cart-item"
        name="Citrus Bag"
        type="bags"
      >
        <div
          className="cart-item-img-container"
        >
          <img
            alt={
              Object {
                "name": "Citrus Bag",
                "price": 45,
                "quantity": 1,
                "source": "citrus_bag.png",
                "type": "bags",
              }
            }
            className="cart-item-img"
            src="citrus_bag.png"
          />
        </div>
        <div
          className="cart-item-side"
        >
          <div
            className="cart-item-info"
          >
            <div
              className="cart-item-name"
            >
              Citrus Bag
            </div>
            <div
              className="cart-item-price"
            >
              $45.00
            </div>
          </div>
          <div
            className="cart-item-quantity-container"
          >
            <button>
              -
            </button>
            <input
              className="cart-item-quantity-input"
              onInput={[MockFunction]}
              type="number"
              value={1}
            />
            <button>
              +
            </button>
          </div>
        </div>
      </div>
      <div
        className="cart-item"
        name="Bowler"
        type="hats"
      >
        <div
          className="cart-item-img-container"
        >
          <img
            alt={
              Object {
                "name": "Bowler",
                "price": 25,
                "quantity": 3,
                "source": "bowler.jpg",
                "type": "hats",
              }
            }
            className="cart-item-img"
            src="bowler.jpg"
          />
        </div>
        <div
          className="cart-item-side"
        >
          <div
            className="cart-item-info"
          >
            <div
              className="cart-item-name"
            >
              Bowler
            </div>
            <div
              className="cart-item-price"
            >
              $25.00
            </div>
          </div>
          <div
            className="cart-item-quantity-container"
          >
            <button>
              -
            </button>
            <input
              className="cart-item-quantity-input"
              onInput={[MockFunction]}
              type="number"
              value={3}
            />
            <button>
              +
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="cart-total"
    >
      Total: $
      120.00
    </div>
    <button
      className="checkout"
    >
      Checkout
    </button>
  </div>
</div>
`;
